src/main/java/ai/elimu/util/ChecksumHelper.java

```diff
@@ -0,0 +1,57 @@
+package ai.elimu.util;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.math.BigInteger;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import org.apache.log4j.Logger;
+
+/**
+ * Based on https://github.com/CyanogenMod/android_packages_apps_CMUpdater/blob/cm-10.2/src/com/cyanogenmod/updater/utils/MD5.java
+ */
+public class ChecksumHelper {
+    
+    private static final Logger logger = Logger.getLogger(ChecksumHelper.class);
+
+    public static String calculateMD5(byte[] bytes) {
+        logger.info("calculateMD5");
+        
+        InputStream inputStream = new ByteArrayInputStream(bytes);
+        
+        MessageDigest messageDigest;
+        try {
+            messageDigest = MessageDigest.getInstance("MD5");
+        } catch (NoSuchAlgorithmException ex) {
+            logger.error(null, ex);
+            return null;
+        }
+
+        byte[] buffer = new byte[8192];
```

Comment:
Can you add an explanation of what 8192 is? And why this number?

Reply:
It's an 8 MB buffer. We will be uploading some APK files that are hundreds of megabytes in size, so a larger buffer would probably speed things up.